llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -764,11 +764,18 @@ void AsmPrinter::emitGlobalVariable(const GlobalVariable *GV) {
 
   const DataLayout &DL = GV->getDataLayout();
   uint64_t Size = DL.getTypeAllocSize(GV->getValueType());
+  if (GV->isTagged())
+    Size = alignTo(Size, 16);
```

Comment:
Wouldn't we need to add padding to the initializer as well to replace the code you're removing in AArch64GlobalsTagging.cpp? I think in the usual case (non-bss) all this would do is change the value of the st_size field in the symbol.

Reply:
We don't need the padding to be initialized to anything in particular, the code shouldn't use it (other than its tag memory)

Reply:
But isn't the padding necessary to prevent another non-MTE global from being placed there? Aligning st_size won't cause that to happen, st_size is almost unused these days.

Reply:
Good catch.